driver/main.ml

```diff
@@ -152,6 +159,16 @@ let main () =
   try
     readenv ppf Before_args;
     Arg.parse Options.list anonymous usage;
+    if !output_name <> None && !compile_only &&
+          List.length !process_thunks > 1 then
```

Comment:
@damiendoligez 
Oups, this check will fail with the fix that we did yesterday, since `-cclib` will increase the length of `process_thunks`. But there was already a bug before, since `ocamlc -c toto.ml -o toto.cmo toto.a` will also be rejected since `toto.a` is also in `process_thunks`.

I think we should remove the functions in `process_thunks`, and use a type with multiple constructors, so that we can be more precise here.